### PR TITLE
Add SetPixel as separate r g b values

### PR DESCRIPTION
**Pull request by LwServices**
## Description
- Improve SetPixel implementation to have separate RGB values.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [x] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nanoframework/nanoFramework.IoT.Device?pullRequest=220) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nanoframework/nanoFramework.IoT.Device?pullRequest=220) before we can accept your contribution.<br/><br/>:x: LwServices [sign now](https://cla.dotnetfoundation.org/nanoframework/nanoFramework.IoT.Device?pullRequest=220)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nanoframework/nanoFramework.IoT.Device?pullRequest=220) it.</sub>

File: devices/Ws28xx/BitmapImage.cs [hunk 1]
```diff
@@ -54,7 +54,17 @@ protected BitmapImage(byte[] data, int width, int height, int stride)
...
+
+        /// <summary>
+        /// Sets pixel at specific position
+        /// </summary>
+        /// <param name="x">X coordinate of the pixel</param>
+        /// <param name="y">Y coordinate of the pixel</param>
+        /// <param name="r">Red color</param>
+        /// <param name="g">Green color</param>
+        /// <param name="b">Blue color</param>
+        public abstract void SetPixel(int x, int y, ushort r, ushort g, ushort b);
```
Comment (Ellerbach) [inline devices/Ws28xx/BitmapImage.cs:67]: why ushort and not byte?

**Review by Ellerbach — COMMENTED** [review]
Thanks for adding this function. As each component of color are byte, why have the function with ushort and not just byte?

File: devices/Ws28xx/BitmapImage.cs (hunk 1 above)
Comment (LwServices) [reply devices/Ws28xx/BitmapImage.cs:67]: I change to byte and add an example

**Comment by josesimoes** [comment]
@LwServices don't mind the last message from nfbot about the build failing. It was a glitch with one of the build tools.

**Comment by josesimoes** [comment]
@LwServices please rebase your PR and fix the merge conflicts with the versions.

**Comment by LwServices** [comment]
@josesimoes rebase is done

**Comment by josesimoes** [comment]
@Ellerbach are you good with this PR?

**Review by Ellerbach — APPROVED** [review]
Thanks for this addition!

**Pull request merged**